### ISSUE TITLE
unable to build texlive: fakeroot segmentation fault

### ISSUE THREAD

**Issue by algent-al** [getsolus/packages#330]
### Summary

When trying to build `texlive` I am getting this error.

```
 ⮞  Now starting build of package
[HISTORY] Release not consistent
[Info] Building texlive-20210325
/usr/bin/fakeroot: line 178: 59748 Segmentation fault      (core dumped) FAKEROOTKEY=$FAKEROOTKEY LD_LIBRARY_PATH="$PATHS" LD_PRELOAD="$FAKEROOT_LIB" "$@"
 🕱  Failed to build packages
```

### Steps to reproduce

1. cd to you packagin dir.
2. Run following commands:
```
make texlive.clone
cd texlive
make bump
make
```

### Expected result

Package built succesfully.

### Actual result

Failed to build packages

### Environment

- [x] Is system up to date?

### Repo

Unstable

### Desktop Environment

Budgie

### System details

OS: Solus x86_64
Host: CELSIUS W530
DE: Budgie 10.8
CPU: Intel Xeon E3-1271 v3 (8) @ 4.000GHz
GPU: NVIDIA Quadro M2000
Memory: 15923MiB 

### Other comments

Assigning to @ReillyBrogan since he wanted to take a look :D

Who ever want to try this, can use these new source links.
```
source     :
    - https://mirrors.tuna.tsinghua.edu.cn/tex-historic-archive/systems/texlive/2021/texlive-20210325-source.tar.xz : 7aefd96608d72061970f2d73f275be5648ea8ae815af073016d3106acc0d584b
    - https://mirrors.tuna.tsinghua.edu.cn/tex-historic-archive/systems/texlive/2021/texlive-20210325-texmf.tar.xz : ff12d436c23e99fb30aad55924266104356847eb0238c193e839c150d9670f1c
```

**Comment by joebonrichie** [comment]
Also fails with `rootlesskit`

```
~/solus/texlive $ ~/rootlesskit/bin/rootlesskit --copy-up=/var/cache/eopkg ~/ypkg/ypkg-build package.yml
[Warning] ypkg-build should be run via fakeroot, not as real root user
[Info] Building texlive-20210325
[Source] Fetching: https://mirrors.tuna.tsinghua.edu.cn/tex-historic-archive/systems/texlive/2021/texlive-20210325-source.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 52.2M  100 52.2M    0     0  5412k      0  0:00:09  0:00:09 --:--:-- 8466k
[Source] Fetching: https://mirrors.tuna.tsinghua.edu.cn/tex-historic-archive/systems/texlive/2021/texlive-20210325-texmf.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3313M  100 3313M    0     0  8004k      0  0:07:03  0:07:03 --:--:-- 7072k
[rootlesskit:child ] error: command [/home/ninya/ypkg/ypkg-build package.yml] exited: signal: segmentation fault (core dumped)
[rootlesskit:parent] error: child exited: exit status 255
~/solus/texlive $ cat package.yml 
~/solus/texlive $ fakeroot ~/ypkg/ypkg-build package.yml
[Info] Building texlive-20210325
/usr/bin/fakeroot: line 178: 573811 Segmentation fault      (core dumped) FAKEROOTKEY=$FAKEROOTKEY LD_LIBRARY_PATH="$PATHS" LD_PRELOAD="$FAKEROOT_LIB" "$@"
```

**Comment by FriesischScott** [comment]
This seems to be related to the size of the tarball(s). The `texmf` tarball is is over 3GB.

This can be reproduced with any large tarball. Using `nerd-fonts` as an example this `package.yml`

```yaml
name       : nerd-fonts
version    : 3.0.2
release    : 1
source     :
    - https://github.com/ryanoasis/nerd-fonts/archive/refs/tags/v3.0.2.tar.gz : 3b18b305451b0b22cb6484197f8029efe54e5c459e8ed3adbb0fb1697c0c13a6
license    : License
summary    : Summary
description: Description
build      : |
    echo "build"
install    : |
    echo "install"
```
will segfault the same as the texlive build.

**Comment by ReillyBrogan** [comment]
Resolved by b35319c48c9ea5f39925a4dc3efc9059465d9e90

**Comment by ReillyBrogan** [comment]
> Also fails with `rootlesskit`

Because it's not an issue with fakeroot, that was a red herring. Python was segfaulting at `hashlib.sha256()` when computing the hash of the source archive. Presumably this is due to some incompatibility with a current lib (libcrypto is my guess) or some bug that has likely been resolved in current python versions. Or perhaps we need to fuck with buffer sizes or something. 

ORRRRR we could just call out to the `sha256sum` binary which is probably faster anyway since that uses SHA256 CPU instructions and this Python2 code needs to diaf anyway.
